### PR TITLE
kdf: add PIN verification key KDF to providers

### PR DESCRIPTION
**Pull request by paulidale**
This is in pem/pvkfmt.c currently.  Since it is a KDF, it is better located in a provider -- in this case legacy since it is only used if RC4 is available.

~This PR is based on #15967, will need to be rebased after that is merged.~

The existing test cases exercise this code sufficiently.

- [x] documentation is added or updated
- [ ] tests are added or updated

### CODE REVIEW

**Comment by paulidale** [comment]
Rebased to current master

**Comment by paulidale** [comment]
Review required.

**Comment by paulidale** [comment]
Was kind of hoping for a 3.0.1/3.1 decisions.  Master feels fine for this change.

**Comment by paulidale** [comment]
Ping for review

**Comment by paulidale** [comment]
Rebased to master, still awaiting review.

**Review by t8m — CHANGES_REQUESTED** [review]
Some things to consider. Also should there be a CHANGES.md entry? We have it for other KDFs.
Also, should we have some test cases in evp tests?

**Comment by paulidale** [comment]
Good feedback @t8m, I'll address net week.

File: providers/implementations/kdfs/pvkkdf.c [hunk 1]
```diff
@@ -0,0 +1,223 @@
...
+        ERR_raise(ERR_LIB_PROV, PROV_R_MISSING_PASS);
+        return 0;
+    }
+
+    if (ctx->salt == NULL) {
+        ERR_raise(ERR_LIB_PROV, PROV_R_MISSING_SALT);
+        return 0;
+    }
+
+    md = ossl_prov_digest_md(&ctx->digest);
```
[the remark this answers is not in the document]
Comment (paulidale) [reply providers/implementations/kdfs/pvkkdf.c:134]: I think that `ossl_prov_digest_md()` should be doing the proper error reporting.  It is used in quite a number of places.

File: doc/man7/OSSL_PROVIDER-legacy.pod [hunk 2]
```diff
@@ -87,6 +87,8 @@ Disabled by default. Use I<enable-rc5> config option to enable.
 
 =item PBKDF1
 
+=item PVK KDF
```
Comment (levitte) [inline doc/man7/OSSL_PROVIDER-legacy.pod:90]: Drop the space

**Comment by paulidale** [comment]
Feedback addressed, test cases added.

**Review by t8m — APPROVED** [review]

**Comment by paulidale** [comment]
Merged to master, thanks for the reviews.

**Pull request closed without merging**